device/core/src/client.ts

```diff
@@ -731,6 +503,31 @@ export class Client extends EventEmitter {
     return typeof this._transport[fnName] === 'function';
   }
 
+  private _closeTransport(closeCallback: (err?: Error, result?: any) => void): void {
+    const onDisconnected = (err?: Error, result?: any): void => {
+      /*Codes_SRS_NODE_DEVICE_CLIENT_16_056: [The `close` method shall not throw if the `closeCallback` is not passed.]*/
+      /*Codes_SRS_NODE_DEVICE_CLIENT_16_055: [The `close` method shall call the `closeCallback` function when done with either a single Error object if it failed or null and a results.Disconnected object if successful.]*/
+      safeCallback(closeCallback, err, result);
+    };
+
+    if (this._sasRenewalTimeout) {
+      clearTimeout(this._sasRenewalTimeout);
+    }
+
+    this._disableC2D(() => {
+      if (this._isImplementedInTransport('disconnect')) {
```

Comment:
>(this._isImplementedInTransport('disconnect')) [](start = 9, length = 46)

My HTTP provisioning transports has empty implementations for connect and disconnect rather than relying on the outer code to take care of it.  That way there's a single transport interface that always operates the same way.

Is there any value in updating my transports to have multiple interfaces depending on the need to connect/disconnect?  My gut says that adding that interface doesn't help me at all, but I"d like to hear your ideas.

Reply:
yeah those tests are remnants of not having type safety and proper interfaces. I wish all transports implemented the same interface and would throw clear NotImplementedErrors themselves. probably a fix for later though, the scope of this PR is large enough already.

---
In reply to: [145241601](https://github.com/Azure/azure-iot-sdk-node/pull/109#discussion_r145241601) [](ancestors = 145241601)